### ISSUE TITLE
Moving cursor and scrolling skyrockets CPU usage

### ISSUE THREAD

**Issue by ZivaUrbancic** [microsoft/vscode#216482]
Type: <b>Performance Issue</b>

I noticed that whenever I use vscode (even at start up), the CPU usage skyrockets. After observing what action triggers it, I noticed that even movement of the cursor and scrolling can be enough. Find the screen recording attached. I was able to reproduce this after restarting both vscode and my computer, while editing files and when no tabs are opened, after disabling all extentions ... I also tested to see that it doesn't happen with other apps I run.

Steps to reproduce:
1. Open vscode
2. Move cursor and observe CPU usage.

VS Code version: Code 1.90.1 (611f9bfce64f25108829dd295f54a6894e87339d, 2024-06-11T21:02:43.666Z)
OS version: Linux x64 6.5.0-35-generic snap
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8665U CPU @ 1.90GHz (8 x 3084)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|2, 1, 1|
|Memory (System)|15.24GB (11.56GB free)|
|Process Argv|--force-user-env --no-sandbox --use-gl=angle --use-angle=swiftshader --crash-reporter-id 630ad90e-fe8b-4d26-971c-a5cef3ccb5ee|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|ubuntu|
|XDG_CURRENT_DESKTOP|Unity|
|XDG_SESSION_DESKTOP|ubuntu|
|XDG_SESSION_TYPE|wayland|
</details><details>
<summary>Process Info</summary>

```
CPU %	Mem MB	   PID	Process
    0	   172	 19737	code main
    0	    31	 19739	   zygote
    0	    31	 19740	   zygote
    0	   250	 19811	window [1] (● jdshglshglkfsg • Untitled-1 - Monoaminergic - Visual Studio Code)
    0	   109	 19874	     window
    0	    94	 20559	window [2] (Issue Reporter)
    0	   172	 19764	   gpu-process
    0	    62	 19774	   utility-network-service
    0	   109	 19849	extensionHost [1]
    0	   109	 19899	shared-process
    0	     0	 20598	     /bin/sh -c /usr/bin/ps -ax -o pid=,ppid=,pcpu=,pmem=,command=
    0	     0	 20599	       /usr/bin/ps -ax -o pid=,ppid=,pcpu=,pmem=,command=
    0	    78	 19900	fileWatcher [1]
```

</details>
<details>
<summary>Workspace Info</summary>

```
|  Window (● jdshglshglkfsg • Untitled-1 - Monoaminergic - Visual Studio Code)
|    Folder (Monoaminergic): 324 files
|      File types: png(131) p(68) ipynb(20) pdf(13) py(9) md(6) pyc(6) sty(4)
|                  zip(4) jpg(3)
|      Conf files:;
```

</details>
Extensions: none<details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscoreces:30445986
vscod805cf:30301675
binariesv615:30325510
vsaa593cf:30376535
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyone:30548225
962ge761:30959799
pythongtdpath:30769146
welcomedialog:30910333
pythonidxpt:30866567
pythonnoceb:30805159
asynctok:30898717
pythontestfixt:30902429
pythonregdiag2:30936856
pythonmypyd1:30879173
h48ei257:31000450
pythontbext0:30879054
accentitlementst:30995554
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
jchc7451:31067544
chatpanelt:31048053
dsvsc021:30996838
da93g388:31013173
pythoncenvpt:31062603
a69g1124:31058053
dvdeprecation:31068756
pythonprt:31056678
dwnewjupyter:31046869
2f103344:31071589
legacy_priority:31077528

```

</details>

**Comment by KristinnVikar** [comment]
Was just about to open a similar issue, I'm running into this exact same problem, also running Ubuntu

**Comment by IanBurwell** [comment]
I am having the same issue with VSCode 1.90.1 on Ubuntu 22.04:
```
Version: 1.90.1
Commit: 611f9bfce64f25108829dd295f54a6894e87339d
Date: 2024-06-11T21:02:43.666Z
Electron: 29.4.0
ElectronBuildId: 9593362
Chromium: 122.0.6261.156
Node.js: 20.9.0
V8: 12.2.281.27-electron.0
OS: Linux x64 6.5.0-35-generic snap
```

**Comment by Pedro-Magalhaes** [comment]
seems to be releated the fix to #212494 when using snap

**Comment by leandrosa** [comment]
It's happing to me too. I also using Ubuntu 22.04. First, I thought it could be related with some extension and I tested with an empty profile (without extensions) and the problem keeps.
```
Version:          Code 1.90.1 (611f9bfce64f25108829dd295f54a6894e87339d, 2024-06-11T21:02:43.666Z)
OS Version:       Linux x64 6.5.0-35-generic
CPUs:             Intel(R) Core(TM) i5-8350U CPU @ 1.70GHz (8 x 1199)
Memory (System):  15.46GB (6.96GB free)
```

**Comment by peacefulotter** [comment]
Same thing for me, scrolling uses 80-100% of all my CPU cores and VSCode feels very laggy in general

**Comment by shelvesdragon** [comment]
As @Pedro-Magalhaes mentioned, this is likely caused by the temporary fix in version 1.90.1 on the snap package that switched the OpenGL backend to software emulation in order to prevent the crashes reported with version 1.90.0 in #212494 . There currently is a candidate build intended to be included in release 1.90.2 mentioned in this comment https://github.com/microsoft/vscode/issues/212494#issuecomment-2168893148 looking for feedback. Otherwise, your options for now are mentioned in this comment: https://github.com/microsoft/vscode/issues/212494#issuecomment-2168072242
Hope this helps!

**Comment by peacefulotter** [comment]
Reverting to 1.90.0 with `snap revert code` indeed fixed it, thanks!
(you can check you vscode version with `code --version`)

**Comment by TheRealSmeddy** [comment]
I was going to open the same issue, lately the cpu usage is annoyingly high
```
Version:          Code 1.90.1 (611f9bfce64f25108829dd295f54a6894e87339d, 2024-06-11T21:02:43.666Z)
OS Version:       Linux x64 6.5.0-35-generic
CPUs:             11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz (8 x 1843)
Memory (System):  23.19GB (14.41GB free)
```

**Comment by kkeri** [comment]
Same issue here.

~~~
Version: 1.90.1
Commit: 611f9bfce64f25108829dd295f54a6894e87339d
Date: 2024-06-11T21:02:43.666Z
Electron: 29.4.0
ElectronBuildId: 9593362
Chromium: 122.0.6261.156
Node.js: 20.9.0
V8: 12.2.281.27-electron.0
OS: Linux x64 6.5.0-35-generic snap
~~~

**Comment by Pedro-Magalhaes** [comment]
Snap just updated vscode to 1.90.2 for me and the problems seems to be fixed for me.

**Comment by deepak1556** [comment]
Merging to https://github.com/microsoft/vscode/issues/212494, should be fixed in `1.90.2`
